pkg/virt-controller/watch/vm_test.go

```diff
@@ -1125,29 +1125,56 @@ var _ = Describe("VirtualMachine", func() {
 					Name: "dv1",
 				},
 			})
-
-			existingDataVolume, _ := watchutil.CreateDataVolumeManifest(virtClient, vm.Spec.DataVolumeTemplates[0], vm)
-
-			existingDataVolume.Namespace = "default"
-			existingDataVolume.Status.Phase = cdiv1.Succeeded
+			vm.Spec.Running = nil
+			vm.Spec.RunStrategy = &runStrategy
 
 			vm, err := virtFakeClient.KubevirtV1().VirtualMachines(vm.Namespace).Create(context.TODO(), vm, metav1.CreateOptions{})
 			Expect(err).To(Succeed())
+
+			shouldExpectDataVolumeCreation(vm.UID, map[string]string{"kubevirt.io/created-by": string(vm.UID)}, nil, pointer.P(0))
+
 			addVirtualMachine(vm)
+			sanityExecute(vm)
```

Comment:
@acardace  We should not use multiple `sanityExecute` in one test case. We should split this

Reply:
Unfortunately to properly test this situation I cannot split it. Unless I call `updateStatus` before calling `sanityExecute`.

Reply:
> @acardace We should not use multiple `sanityExecute` in one test case. We should split this

Why? Is is a technical limitation of `sanityExecute()`? When seeing this, I was actually thinking "great, we don't have enough tests that call execute multiple times"!
In real life, and regardless of the controller, `execute()` often gets called multiple times on the same object, doing things one step at a time. If we want to test a multi-step operation, we need to call it multiple times.
For example, a migration controller test might want to ensure a migration goes from pending to running after 4 `execute()` calls. Just testing that pending -> scheduling works, then in a different test scheduling -> scheduled might miss a bug where, for example, the transition from pending to scheduling broke something that will prevent going to scheduled.
@xpivarc do you disagree?

Reply:
I think it's more because of test readability and maintainability, but @jean-edouard you also have a point, calling `execute` once is less powerful in terms of what you can exercise in a test and this PR is a good example.

I think you're both right, maybe we just need to find a balance.